packages/elements/src/components/progress-bar/progress-bar.component.ts

```diff
@@ -0,0 +1,146 @@
+import { MutationTestMetricsResult } from 'mutation-testing-metrics';
+import { RealTimeElement } from '../real-time-element';
+import { PropertyValues, html, nothing } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { tailwind } from '../../style';
+
+interface ProgressBarMetrics {
+  killed: number;
+  survived: number;
+  combined: number;
+  [key: string]: number;
+}
+
+@customElement('mte-progress-bar')
+export class ProgressBar extends RealTimeElement {
+  public static styles = [tailwind];
+
+  @property({ attribute: false })
+  public rootModel: MutationTestMetricsResult | undefined;
+
+  @property({ attribute: false })
+  public metrics: ProgressBarMetrics | undefined;
+
+  @property({ attribute: false })
+  public total = 0;
+
+  @property({ attribute: false })
+  public visible = false;
+
+  #observer: IntersectionObserver | undefined;
+  #shouldBeSmall = false;
+
+  public constructor() {
+    super();
+  }
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+
+    this.#observer = new window.IntersectionObserver(([entry]) => {
+      if (entry.isIntersecting) {
+        this.#shouldBeSmall = false;
+      } else {
+        this.#shouldBeSmall = true;
+      }
+
+      this.requestUpdate();
+    });
```

Comment:
Could you add a comment here? What are you doing and why?

Reply:
It's checking whether the progress-bar is visible or not (intersects with the `visible` window), but a comment doesn't hurt 😃